distributed/utils_test.py

```diff
@@ -1800,8 +1809,22 @@ def check_instances():
     assert all(c.status == Status.closed for c in SpecCluster._instances), list(
         SpecCluster._instances
     )
-    SpecCluster._instances.clear()
+    wait_profiler()
+    gc.collect()
+    if Scheduler._instances:
+        s = next(iter(Scheduler._instances))
+        import objgraph
+
+        def ignore(obj):
+            return (
+                # ignore bound methods temporarily
+                not (inspect.ismethod(obj) and obj.__self__ is s))
```

Comment:
When not ignoring bound methods, `SchedulerState` methods appear to be holding onto a `Scheduler` reference. I still need to figure this out.

![scheduler](https://user-images.githubusercontent.com/3530212/166970957-79cc59c5-151f-49d6-bc4e-4f6cee8de719.png)

Reply:
The `transitions_table` member needed to cleared.